MIGRATING.md

```diff
@@ -0,0 +1,92 @@
+
+### Breaking Changes
+
+Following is a list of changes that break backward compatibility in version 2.0.0.
+
+- `com.okta.sdk.client.Client`
+   - Renamed `createRule` to `createGroupRule`
+   - Renamed `getRule` to `getGroupRule`
+   - Renamed `listRules` to `listGroupRules`
+   
+- `com.okta.sdk.resource.log.LogEventList` 
+   - `getLogs(String until, String since, String filter, String q, String sortOrder)`
+   Signature changed to `getLogs(Date since, Date until, String filter, String q, String sortOrder)`
+   
+- `com.okta.sdk.resource.group.GroupList` 
+   - `listGroups(String q, String filter, String expand)`
+   Signature changed to `listGroups(String q, String filter)`
+
+- `com.okta.sdk.resource.application.SwaThreeFieldApplicationSettingsApplication`
+   - `getTargetUrl` Renamed to `getTargetURL`
+   - `setTargetUrl` Renamed to `setTargetURL`
+
+- `com.okta.sdk.resource.group.Group`
+   - `getType` Return type changed from `String` to `com.okta.sdk.resource.group.GroupType`
+   
+- `com.okta.sdk.resource.group.rule.GroupRule`
+   - `delete(Boolean removeUsers)` Signature changed to `delete()`
+   - Removed `getAllGroupsValid` and `setAllGroupsValid` (property `allGroupsValid` is removed)
+   - Removed `getEmbedded` (property `_embedded` is removed)
+
+- `com.okta.sdk.resource.log.LogAuthenticationContext`
+   - `getCredentialProvider` Return type changed from `List<com.okta.sdk.resource.log.LogCredentialProvider>` to `com.okta.sdk.resource.log.LogCredentialProvider`
+   - `getCredentialType` Return type changed from `List<com.okta.sdk.resource.log.LogCredentialType>` to `com.okta.sdk.resource.log.LogCredentialType`
+
+- `com.okta.sdk.resource.policy.PolicyRule`
+   - Removed `setId` (property `id` is `read-only`)
+
+- `com.okta.sdk.resource.user.factor.VerifyFactorRequest`
+   - Removed `getTokenLifetimeSeconds` & `setTokenLifetimeSeconds` (property `tokenLifetimeSeconds` is removed)
+
+- `com.okta.sdk.resource.user.Role`
+   - `getType` Return type changed from `String` to `com.okta.sdk.resource.role.RoleType`
+   - `setType` Param type changed from `String` to an enum `com.okta.sdk.resource.role.RoleType`
+
+- All `*Factor` classes Renamed to `*UserFactor`
+
+- `com.okta.sdk.resource.user.User`
+   - Renamed `addGroupTarget` to `addGroupTargetToRole`
+   - Renamed `addRole` to `assignRole`
+   - Renamed `listGroupTargetsForRole` to `listGroupTargets`
+   - Renamed `removeGroupTargetFromRole` to `removeGroupTarget`
+   - Renamed `addFactor` to `enrollFactor`
+   - Renamed `listRoles`to `listAssignedRoles`
+   - Removed `forgotPassword` (use `resetPassword` instead)
+   - Removed `endAllSessions` (use `clearSessions` instead)
+   - `listSupportedFactors` Return type changed from `com.okta.sdk.resource.user.factor.FactorList` to `com.okta.sdk.resource.user.factor.UserFactorList`
+   - `expirePassword` Return type changed from `com.okta.sdk.resource.user.TempPassword` to `com.okta.sdk.resource.user.User`
```

Comment:
Was `TempPassword` removed too?

Reply:
Nope, `TempPassword` remains.

Reply:
IIRC there's a new method to cover the temp pass use case. You could mention that.

Reply:
The original intention of this note is to explain the changes to `expirePassword` return type.  I feel there is no need to explain the `TempPassword` use case here.